### PR TITLE
"onComputer" implementation of GesturePod

### PR DESCRIPTION
**Pull request by ShishirPatil**

### CODE REVIEW

File: Applications/GesturePod/README.md [hunk 1]
```diff
@@ -8,7 +8,7 @@ for instance a mobile phone. Watch the [GesturePod video here](https://1drv.ms/u
 
 In this directory, we provide resources for
   1. Creating a GesturePod of your own based on the [Arduino MKR1000](https://store.arduino.cc/usa/arduino-mkr1000), in the `onMKR1000` directory, and
-  2. Running a simulation of the gesture recognition algorithm on your desktop in the `onWorkStation` directory. 
+  2. Running a simulation of the gesture recognition algorithm on your desktop in the `onComputer` directory.
```
Comment (metastableB) [inline Applications/GesturePod/README.md]: The bare-bones gesture recognition pipeline using saved data. (please don't use the simulation word)

File: Applications/GesturePod/onComputer/README.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
+GesturePod *(Simulation)*
```
Comment (metastableB) [inline Applications/GesturePod/onComputer/README.md]: GesturePod Gesture Recognition Pipeline

File: Applications/GesturePod/onComputer/README.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+## About
+
+GesturePod is a plug-and-play, gesture recognition device. 
+
+As the GesturePod involves hardware, this is a quick simulation of GesturePod on your system. We have stored sensor data collected from the GesturePod. This pre-stored sensor data is used for predictions. You can now play around with GesturePod on your system!
+
+## Quick Start
+
+1. Installation requires [Git](https://git-scm.com/), [make](https://www.gnu.org/software/make/), and [g++](https://gcc.gnu.org/).
```
Comment (metastableB) [inline Applications/GesturePod/onComputer/README.md]: list please

File: Applications/GesturePod/onComputer/README.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+2. Clone the repo and navigate to this folder
```
Comment (metastableB) [inline Applications/GesturePod/onComputer/README.md]: directory

**Comment by metastableB** [comment]
@harsha-simhadri  There are snippets of data being released with this PR. Please have a discussion with Shishir.

File: Applications/GesturePod/README.md (hunk 1 above)
Comment (ShishirPatil) [reply Applications/GesturePod/README.md]: replaced with Gesture Recognition Pipeline

File: Applications/GesturePod/onComputer/README.md [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+- [About](#about)
+- [Quick Start](#quick-start)
+- [Docs](#docs)
+- [Also](#also)
+
+## About
+
+GesturePod is a plug-and-play, gesture recognition device. 
+
+As the GesturePod involves hardware, this implements GesturePod's gesture recognition pipeline on your system. We have stored sensor data collected from the GesturePod. This pre-stored sensor data is used for gesture recognition. You can now play around with GesturePod on your system!
```
Comment (harsha-simhadri) [inline Applications/GesturePod/onComputer/README.md]: @ShishirPatil on your "system"? you mean PC or arduino?

File: Applications/GesturePod/onComputer/README.md [hunk 6]
```diff
@@ -0,0 +1,46 @@
+GesturePod *(Gesture Recognition Pipeline)*
```
Comment (harsha-simhadri) [inline Applications/GesturePod/onComputer/README.md]: @ShishirPatil could you please check license info here

File: Applications/GesturePod/onComputer/README.md [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+	```
+3. Compile and build the code
+	```
+	make
+	```
+4. Run
+	```
+	./gesturepodsim
+	```
+5. By default the program runs for data that has the following gesture: *Double Taps*.
```
Comment (harsha-simhadri) [inline Applications/GesturePod/onComputer/README.md]: @ShishirPatil  So, this program is for taps or all gestures? The readme is confusing here. Why do you have a separate file for taps and a common data file?

File: Applications/GesturePod/onComputer/README.md (hunk 5 above)
Comment (ShishirPatil) [reply Applications/GesturePod/onComputer/README.md]: PC. Clarified in the latest commit.

File: Applications/GesturePod/onComputer/README.md (hunk 6 above)
Comment (ShishirPatil) [reply Applications/GesturePod/onComputer/README.md]: We are not dependent on any external libraries for this directory. No code snippets used either. We have ownership on every line of code, for this directory.

File: Applications/GesturePod/onComputer/README.md (hunk 7 above)
Comment (ShishirPatil) [reply Applications/GesturePod/onComputer/README.md]: The program runs for the "gesture" contained in the data file present in ```./data/``` directory. Right now, the only gesture contained is Taps. Readme updated to reflect this, and articulate it better.

File: Applications/GesturePod/onComputer/README.md [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+- [About](#about)
+- [Quick Start](#quick-start)
+- [Docs](#docs)
+- [Also](#also)
+
+## About
+
+GesturePod is a plug-and-play, gesture recognition device. 
+
+As the GesturePod involves hardware, this implements GesturePod's gesture recognition pipeline on your computer system. We have stored sensor data collected from the GesturePod. This pre-stored sensor data is used for gesture recognition. You can now play around with GesturePod on your computer!
```
Comment (metastableB) [inline Applications/GesturePod/onComputer/README.md]: Could you remove the '!' marks.

Also, change this to the following.

> This repository isolates the gesture recognition pipeline that is part of the `Arduino MKR1000` implementation. All additional modules, for instance, the Bluetooth communication module, that interfaces with the IMU, and so forth are removed. Data is read from a provided text file and prediction is performed. This repository is intended to be used as a playground to understand the gesture recognition pipeline as well as a bare-minimum implementation of gesture recognition that can be used  as a basis for porting GesturePod to devices other than `MKR1000`.

File: Applications/GesturePod/onComputer/README.md [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+
+## Quick Start
+
+1. Installation requires 
+	- [git](https://git-scm.com/),
```
Comment (metastableB) [inline Applications/GesturePod/onComputer/README.md]: ??

File: Applications/GesturePod/onComputer/README.md [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+	```
+3. Compile and build the code
+	```
+	make
+	```
+4. Run
+	```
+	./gesturepodsim
+	```
+5. By default the program runs for data file included in ```./data/```. This data file has the following gesture: *Double Taps*.
```
Comment (metastableB) [inline Applications/GesturePod/onComputer/README.md]: By default, the program reads data from the `./data/FILENAME` file which contains space-separated IMU data collected during double taps.

File: Applications/GesturePod/onComputer/README.md [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+
+	To run the gesture recognition pipeline for different gestures contained in other data file(s), add the data file(s) in ```./data/``` directory and update ```DATA_FILE``` in ```src/main.cpp:23```.
```
Comment (metastableB) [inline Applications/GesturePod/onComputer/README.md]: Don't do this. This is bad.

File: Applications/GesturePod/onComputer/README.md [hunk 12]
```diff
@@ -0,0 +1,46 @@
...
+	make
+	```
+4. Run
+	```
+	./gesturepodsim
+	```
+5. By default the program runs for data file included in ```./data/```. This data file has the following gesture: *Double Taps*.
+
+	To run the gesture recognition pipeline for different gestures contained in other data file(s), add the data file(s) in ```./data/``` directory and update ```DATA_FILE``` in ```src/main.cpp:23```. 
+	An exhaustive data set is released as a part of our [TechReport](https://www.microsoft.com/en-us/research/uploads/prod/2018/05/dataTR_v1.tar-5b058a4590168.gz)
```
Comment (metastableB) [inline Applications/GesturePod/onComputer/README.md]: The dataset used for training GesturePod can be found [here](https://put.link.here/)

File: Applications/GesturePod/onComputer/README.md [hunk 13]
```diff
@@ -0,0 +1,46 @@
...
+
+## Docs
+Checkout our Blog post *Coming Soon..!*
+
+## Also
```
Comment (metastableB) [inline Applications/GesturePod/onComputer/README.md]: Why ? Please remove this. This is part of the top level Readme.

File: Applications/GesturePod/onComputer/README.md [hunk 14]
```diff
@@ -0,0 +1,46 @@
...
+4. Run
+	```
+	./gesturepodsim
+	```
+5. By default the program runs for data file included in ```./data/```. This data file has the following gesture: *Double Taps*.
+
+	To run the gesture recognition pipeline for different gestures contained in other data file(s), add the data file(s) in ```./data/``` directory and update ```DATA_FILE``` in ```src/main.cpp:23```. 
+	An exhaustive data set is released as a part of our [TechReport](https://www.microsoft.com/en-us/research/uploads/prod/2018/05/dataTR_v1.tar-5b058a4590168.gz)
+
+## Docs
```
Comment (metastableB) [inline Applications/GesturePod/onComputer/README.md]: Remove this section. Why is this part of `onComputer` ?

File: Applications/GesturePod/onComputer/README.md (hunk 7 above)
Comment (metastableB) [reply Applications/GesturePod/onComputer/README.md]: You are assuming a lot of things here - a lot can go wrong. As far as I understand, the point of this code-isolation was to help people who want to understand/play-around with the gesture recognition pipeline. There is no need for us to support arbitrary number of data files. I think one data file is enough. 

Some of the assumptions include:
  1. The `./data/` directory will always exist
  2. All files in that directory will contain your data.
  3. All files will be parseble with your parser. 
  4. You will be able to handle arbitrary time-shifts that will happen between files. 

Any of these will not hold true when people start modifying/playing around with these files.

We can do without these complications.

Comment (ShishirPatil) [reply Applications/GesturePod/onComputer/README.md]: Fixed.

File: Applications/GesturePod/onComputer/README.md (hunk 8 above)
Comment (ShishirPatil) [reply Applications/GesturePod/onComputer/README.md]: Updated with minor modifications.

**Pull request merged**